packages/pwa/app/pages/product-list/index.jsx

```diff
@@ -143,91 +115,75 @@ const ProductList = (props) => {
+                        View
+                    </Button>
+                )
+            })
+        } catch {
+            toast({
+                title: formatMessage(
+                    {defaultMessage: '{errorMessage}'},
+                    {errorMessage: API_ERROR_MESSAGE}
+                ),
+                status: 'error'
             })
-            // remove the loading id
-            setWishlistLoading(wishlistLoading?.filter((id) => id !== product.productId))
-        } catch (err) {
-            showError()
+        } finally {
+            setWishlistLoading(wishlistLoading.filter((id) => id !== product.productId))
         }
     }
-
-    const showWishlistItemAdded = (quantity) => {
-        const toastAction = (
-            <Button variant="link" onClick={() => navigate('/account/wishlist')}>
-                View
-            </Button>
-        )
-        showToast({
-            title: formatMessage(
-                {
-                    defaultMessage:
-                        '{quantity} {quantity, plural, one {item} other {items}} added to wishlist'
-                },
-                {quantity}
-            ),
-            status: 'success',
-            action: toastAction
-        })
-    }
-
-    const addItemToWishlist = async (product) => {
+    const removeItemFromWishlist = async (product) => {
         try {
             setWishlistLoading([...wishlistLoading, product.productId])
-            // If product-lists have not loaded we push "Add to wishlist" event to eventQueue to be
-            // processed once the product-lists have loaded.
-            if (!customerProductLists?.loaded) {
-                const event = {
-                    item: {...product, id: product.productId, quantity: 1},
-                    action: 'add',
-                    listType: customerProductListTypes.WISHLIST
-                }
-
-                customerProductLists.addActionToEventQueue(event)
-            } else {
-                const quantity = 1
-                const wishlist = customerProductLists.getProductListPerType(
-                    customerProductListTypes.WISHLIST
-                )
-                await customerProductLists.createCustomerProductListItem(wishlist, {
-                    productId: product.productId,
-                    priority: 1,
-                    quantity,
-                    public: false,
-                    type: 'product'
-                })
-                showWishlistItemAdded(quantity)
-                setWishlistLoading(wishlistLoading?.filter((id) => id !== product.productId))
-            }
-        } catch (err) {
-            showError()
+            await wishlist.removeItemByProductId(product.productId)
+            toast({
+                title: formatMessage({defaultMessage: 'Item removed from wishlist'}),
+                status: 'success',
+                id: product.productId
```

Comment:
Be careful of using the id for toast in PLP since it is possible for the user to add/remove the same item multiple times while the toast is still on screen, which will cause an error about id duplication.
It is better to let chakra take care of the id.

Reply:
good catch, why do we even need to pass id to the toast? i'm going to remove that...